internal/health_checks/health_checks.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package health_checks
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+var healthChecksLogFile = "health_checks_log.txt"
+
+type HealthCheck interface {
+	Name() string
+	RunCheck(logger *log.Logger) error
+}
+
+type HealthCheckRegistry []HealthCheck
+
+func createHealthChecksLogger(logDir string) (*log.Logger, error) {
+	// Make sure the directory exists before writing the file.
+	if err := os.MkdirAll(filepath.Dir(logDir), 0755); err != nil {
+		return nil, fmt.Errorf("failed to create directory for %q: %w", logDir, err)
+	}
+	file, err := os.OpenFile(filepath.Join(logDir, healthChecksLogFile), os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
Add `defer file.Close()`

Reply:
If I add the `defer file.Close()` inside the method `createHealthChecksLogger`, would that affect the logger object binded to file 🤔 ? Mmmmm I'm figuring out what is the best way to add the `file.Close`. Do you have any suggestions ?

Reply:
Sorry I missed that part -- you're right, that file needs to stay open after that function ends. It doesn't look like `log.Logger` has a `Close()` function so you'd have to hold on to the file handle created inside `createHealthChecksLogger` and then close it once the caller is done with it:

```go
func createHealthChecksLogger(logDir string) (*log.Logger, *os.File, error) {
	// ...

	return log.New(file, "", log.Ldate|log.Ltime|log.Lshortfile), &file, nil
}

func (r HealthCheckRegistry) RunAllHealthChecks(logDir string) (map[string]string, error) {
	// ...

	logger, file, err := createHealthChecksLogger(logDir)
	if err != nil {
		return result, err
	}
        defer file.Close()

	// ...
}
```

Reply:
Or you could even return a closer func instead to avoid leaking the file handle:

```go
func createHealthChecksLogger(logDir string) (logger *log.Logger, closer func(), err error) {
	// ...

	return log.New(file, "", log.Ldate|log.Ltime|log.Lshortfile), func() { file.Close() }, nil
}

func (r HealthCheckRegistry) RunAllHealthChecks(logDir string) (map[string]string, error) {
	// ...

	logger, closer, err := createHealthChecksLogger(logDir)
	if err != nil {
		return result, err
	}
        defer closer()

	// ...
}
```